### PR TITLE
Update mafia.js

### PR DESCRIPTION
**Pull request by moonstruckminun**
Major update, promise I tested everything
Not pushing things I wasn't able to test because they required a working server, which includes fixing links for android (RIP userbase).

### CODE REVIEW

File: scripts/mafiachecker.js [hunk 1]
```diff
@@ -351,11 +363,17 @@ function mafiaChecker() {
...
                             if (commandList.indexOf("protect") !== -1)
                                 commonOptional = commonOptional.concat(["protectmsg"]);
+                            if (commandList.indexOf("bomb") !== -1)
+                                commonOptional = commonOptional.concat(["bombMsg"]);
                             if (commandList.indexOf("dayprotect") !== -1)
                                 commonOptional = commonOptional.concat(["dayProtectMsg", "dayProtectType"]);
+                            if (commandList.indexOf("daydistract") !== -1)
+                                commonOptional = commonOptional.concat(["dayDistractMsg", "dayDistractType"]);
+                            if (commandList.indexOf("voteblock") !== -1)
+                                commonOptional = commonOptional.concat(["voteBlockMsg", "voteBlockDuration"]);
```
Comment (moonstruckminun) [inline scripts/mafiachecker.js]: really

Comment (Fuzzysqurl) [inline scripts/mafiachecker.js]: Ignore these but yu have a lot of other error. 

What kind of text editor do you use? Notepad++ has a checker you can add very easily

Comment (moonstruckminun) [inline scripts/mafiachecker.js]: TextWrangler. Are there better Notepad++ alternatives for Mac?

**Comment by Fuzzysqurl** [comment]
Dumb thing won't let me reply to your reply now:

I'd ask jinora what he uses as a text editor. He's about the only other coder that uses Mac that isn't a dead lamperi.

**Comment by moonstruckminun** [comment]
Hopefully he can let me know lol

The 4 violations left are the ones on the checker asking for {}, which isn't necessary, so this is (hopefully) good to go.

**Comment by Fuzzysqurl** [comment]
You missed my comments :(

**Comment by moonstruckminun** [comment]
I literally don't see anything after hound

**Comment by moonstruckminun** [comment]
which commit are they on?

**Comment by Fuzzysqurl** [comment]
First commit

Line 4727 you have an "else" but lack curly braces
Line 4733 and 4736 need ending semicolons

**Comment by moonstruckminun** [comment]
Does else require brackets? I thought it could be used like other conditionals if it was only used for one line of code. Will change it anyway.

**Comment by moonstruckminun** [comment]
I think that covers it, right?

**Comment by Fuzzysqurl** [comment]
Well the else part has a much larger chance to break if someone isn't careful in a revision. 

Also ideally brackets should always be used but we all break that rule (though I've never seen using an "else" without brackets)

**Pull request merged**